fix: issue #795

007qr opened the pull request:
This PR fixes the issue: #795 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Infinite scrolling now exposes loading and error states for fetched pages.
  * Scroll loading is more resilient when the sentinel element is added, removed, or reattached.

* **Bug Fixes**
  * Prevents duplicate or stale page appends during fast updates.
  * Improves handling of empty result pages so pagination ends correctly.
  * Cleans up observer behavior to avoid lingering scroll listeners.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

007qr committed:
fix: issue #795

007qr committed:
fix: issues with reactivity

007qr committed:
fix: removed duplicate imports

thetarnav commented on packages/pagination/src/index.ts:
```diff
@@ -253,64 +253,75 @@ export type _E = JSX.Element;
 
-  const [contents] = createResource(page, fetcher);
+    createComputed(() => {
+        const content = contents()
```
I believe that `contents.latest` was used so that calling `createInfiniteScroll` itself doesn't trigger a suspense.

007qr committed:
fix: reactivity issue with error and loding

007qr committed:
Merge branch 'main' into main

atk reviewed (changes requested):
Please let's discuss the API. I dislike the boolean and that the params need to be wrapped in a function even if they are not supposed to change.

007qr commented on packages/pagination/src/index.ts:
```diff
@@ -246,71 +246,166 @@ declare module "solid-js" {
 
 /**
- * Provides an easy way to implement infinite scrolling.
+ * A SolidJS utility to create an infinite scrolling experience with IntersectionObserver and reactivity.
```
sure will improve this

007qr committed:
Merge branch 'solidjs-community:main' into main

007qr committed:
Merge branch 'solidjs-community:main' into main

007qr committed:
fix: api design

007qr commented:
@atk I have updated the api design. Please check once

007qr committed:
Merge branch 'solidjs-community:main' into main

007qr committed:
fix(pagination): expose loading/error on createInfiniteScroll pages (#795)

  Addresses review feedback on #797: drop the custom PagesAccessor<T> type
  and reuse solid-js's Resource<T> shape inline via Pick, so we're not
  re-creating a type that already exists in solid-js. Attach reactive
  loading/error getters to the pages accessor as documented. Also removes
  the dead _E export and the stacked onCleanup handlers in the loader
  ref-callback.

007qr commented on packages/pagination/src/index.ts:
```diff
@@ -352,72 +352,134 @@ declare module "solid-js" {
 export type _E = JSX.Element;
 
+/** Function-shaped accessor with live .loading / .error properties */
+export type PagesAccessor<T> = (() => T[]) & {
```
I have changed type to use original type now and made some other changes too. Please let me know if you want to discuss the API design any further...

007qr commented on pnpm-workspace.yaml:
Need to remove this

007qr committed:
refactor: removed testing project from pnpm-workspace-yaml file

007qr committed:
chore: add changeset for pagination createInfiniteScroll improvements

007qr committed:
Merge branch 'main' into main

007qr commented:
Hi @thetarnav, Could you please take a look when you get a chance?

davedbase committed:
Merge branch 'main' into main

007qr committed:
Merge branch 'solidjs-community:main' into main
